src/implementations/ScriptFunctions.cpp

```diff
@@ -90,7 +90,11 @@ void ScriptFunctionsHelper::DisplayTextSlot(QWidget* parentWidget, const QString
   popup.setWindowTitle(title);
   // the size readmes are becoming automatically
   popup.resize(492, 366);
-  popup.exec();
+  popup.setWindowModality(Qt::ApplicationModal);
+  popup.show();
+  QEventLoop loop;
+  connect(&popup, SIGNAL(closed()), &loop, SLOT(quit()), Qt::DirectConnection);
+  loop.exec();
```

Comment:
New question. The popup's constructor sets `WA_DeleteOnClose`, but it's stack-allocated here. Is there anything guaranteeing we don't end up with a double-free?

Reply:
I never encountered issues with this when testing but I did not manage to find a clear answer for this in Qt documentation so it's probably better to move the `setAttribute` outside of the constructor.